Doc/library/interpreters.rst

```diff
@@ -0,0 +1,188 @@
+
+    .. method:: destroy()
+
+       Destroy the interpreter. Attempting to destroy the current
+       interpreter results in a `RuntimeError`.
+
+    .. method:: run(self, src_str, /, *, channels=None):
+
+       Run the given source code in the interpreter. This blocks
+       the current thread until done. `channels` should be in
+       the form : `(RecvChannel, SendChannel)`.
+
+RecvChannel Objects
+-------------------
+
+The RecvChannel object represents a recieving channel.
+
+.. class:: RecvChannel(id)
+
+    This class represents the receiving end of a channel.
+
+    .. method:: recv()
+
+        Get the next object from the channel, and wait if
+        none have been sent. Associate the interpreter
+        with the channel.
+
+    .. method:: recv_nowait(default=None)
+
+        Like ``recv()``, but return the default result
+        instead of waiting.
+
+    .. method:: release()
+
+        Release the channel for the current interpreter.
+        By default both ends are released. Releasing an already
+        released end results in a ``ChannelReleasedError`` exception.
+
+    .. method:: close(force=False)
+
+        Close the channel in all interpreters. By default
+        both ends are closed. closing an already closed end
+        results in a ``ChannelClosedError`` exception. Without
+        seeting ``force`` to ``True`` a ``ChannelNotEmptyError``
+        will be returned when a channel with data is closed.
+
+
+SendChannel Objects
+--------------------
+
+The ``SendChannel`` object represents a sending channel.
+
+.. class:: SendChannel(id)
+
+    This class represents the sending end of a channel.
+
+    .. method:: send(obj)
+
+       Send the object ``obj`` to the receiving end of the channel
+       and wait. Associate the interpreter with the channel.
+
+    .. method:: send_nowait(obj)
+
+        Like ``send()`` but return ``False`` if not received.
+
+    .. method:: send_buffer(obj)
+
+       Send the object's buffer to the receiving end of the
+       channel and wait. Associate the interpreter with the
+       channel.
+
+    .. method:: send_buffer_nowait(obj)
+
+       Like ``send_buffer()`` but return ``False`` if not received.
+
+    .. method:: release()
+
+        Release the channel for the current interpreter.
+        By default both ends are released. Releasing an already
+        released end results in a ``ChannelReleasedError`` exception.
+
+    .. method:: close(force=False)
+
+        Close the channel in all interpreters. By default
+        both ends are closed. closing an already closed end
+        results in a ``ChannelClosedError`` exception.
+
+
+This module defines the following global functions:
+
+
+.. function:: is_shareable(obj)
+
+   Return ``True`` if the object's data can be shared between
+   interpreters.
+
+.. function:: create_channel()
+
+   Create a new channel for passing data between interpreters.
+
+.. function:: list_all_channels()
+
+   Return all open channels.
```

Comment:
Does this return a list of Channel objects? If so, I think it would be helpful to mention that here.